### PR TITLE
feat: expose unreal as C-ABI

### PR DESCRIPTION
**Pull request by bruno-garcia**

### CODE REVIEW

File: py/symbolic/unreal.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from symbolic._lowlevel import lib, ffi
+from symbolic.utils import RustObject, rustcall
+
+__all__ = ['Unreal4Crash']
+
+class Unreal4Crash(RustObject):
```
Comment (jan-auer) [inline py/symbolic/unreal.py]: This needs a

```py
    __dealloc_func__ = lib.semaphore_unreal4crash_free
```

File: py/symbolic/unreal.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    @classmethod
+    def from_bytes(cls, buffer):
+        """Parses an Unreal Engine 4 crash"""
+        buffer = bytes(buffer)
+        return cls._from_objptr(rustcall(lib.symbolic_unreal4_crash_from_bytes,
+                                buffer, len(buffer)))
```
Comment (jan-auer) [inline py/symbolic/unreal.py]: In other cases we do `ffi.from_buffer(buffer)` to pass in the buffer. But if it works, it works.

File: py/symbolic/unreal.py (hunk 1 above)
Comment (bruno-garcia) [reply py/symbolic/unreal.py]: good catch. Except u've been doing too much `semaphore` hehe
`symbolic_unreal4_crash_free`

File: py/symbolic/unreal.py (hunk 2 above)
Comment (bruno-garcia) [reply py/symbolic/unreal.py]: Not sure what's the difference here. maybe @mitsuhiko has some insight?

Comment (jan-auer) [reply py/symbolic/unreal.py]: https://cffi.readthedocs.io/en/latest/ref.html#ffi-buffer-ffi-from-buffer
It basically ensures to give you the raw pointer, no matter what you pass in as long as it conforms to the buffer interface.

**Pull request merged**